### PR TITLE
Added temp file in crashreports because git is stupid

### PR DESCRIPTION
**Pull request by BitBaseBit**
Forgot that git always ingnores empty dirs, so the "crashreports" directory was not created from my last PR

### CODE REVIEW

**Comment by JekaMas** [comment]
If a user uses a binary from git release it won't help. Maybe it's better to check for a folder in datadir and make one if needed?

**Comment by BitBaseBit** [comment]
> If a user uses a binary from git release it won't help. Maybe it's better to check for a folder in datadir and make one if needed?

Will do

**Comment by BitBaseBit** [comment]
> If a user uses a binary from git release it won't help. Maybe it's better to check for a folder in datadir and make one if needed?

I think mdbx has exclusive permission on datadir, so I can't create the folder there. I will just put it in the home folder for now.

**Comment by JekaMas** [comment]
> > If a user uses a binary from git release it won't help. Maybe it's better to check for a folder in datadir and make one if needed?
> 
> I think mdbx has exclusive permission on datadir, so I can't create the folder there. I will just put it in the home folder for now.

I was incorrect. One level up from datadir. I just don't remember the flag name, unfortunately.

**Comment by BitBaseBit** [comment]
> > > If a user uses a binary from git release it won't help. Maybe it's better to check for a folder in datadir and make one if needed?
> > 
> > 
> > I think mdbx has exclusive permission on datadir, so I can't create the folder there. I will just put it in the home folder for now.
> 
> I was incorrect. One level up from datadir. I just don't remember the flag name, unfortunately.

Done

**Comment by AlexeyAkhunov** [comment]
Can you please exclude `tests/testdata` submodule update?

File: common/debug/log_panic.go [hunk 1]
```diff
@@ -23,14 +24,14 @@ func prettyTime() string {
...
-func CheckForCrashes() {
-	ex, err := os.Executable()
-	if err != nil {
-		log.Warn(err.Error())
+func CheckForCrashes(datadir string) {
+	crashReportDir = filepath.Join(datadir, "crashreports")
+	if _, err := os.Stat(crashReportDir); os.IsNotExist(err) {
+		os.Mkdir(crashReportDir, 0755)
+	} else if err != nil {
+		log.Error(err.Error())
```
Comment (AlexeyAkhunov) [inline common/debug/log_panic.go:32]: It is better to do something like this (to indicate the location of the error):
```
log.Error("CheckForCrashes", "error", err)
```

**Comment by BitBaseBit** [comment]
> > > > If a user uses a binary from git release it won't help. Maybe it's better to check for a folder in datadir and make one if needed?
> > > 
> > > 
> > > I think mdbx has exclusive permission on datadir, so I can't create the folder there. I will just put it in the home folder for now.
> > 
> > 
> > I was incorrect. One level up from datadir. I just don't remember the flag name, unfortunately.
> 
> Done

You mean this https://github.com/ledgerwatch/erigon/commit/74862ce6074e2cb9ab41cd39f17268ea4fcb7ab4 ?

**Comment by AlexeyAkhunov** [comment]
> You mean this [74862ce](https://github.com/ledgerwatch/erigon/commit/74862ce6074e2cb9ab41cd39f17268ea4fcb7ab4) ?

This PR should not change that submodule. I know submodules are a bit of a pain.

**Comment by BitBaseBit** [comment]
> This PR should not change that submodule. I know submodules are a bit of a pain.

Ah sorry I didn't notice that 'make test' changes the commit of the submodule. I've switched it back.

**Comment by BitBaseBit** [comment]
Hmm I note sure why the CI failed p2p test. I ran it on my macbook (macos 15) and it passed p2p but failed on turbo/snapshotsync : 
--- FAIL: TestSnapshotMigratorStageSyncMode (2.71s)
580     snapshot_builder_test.go:622: mdbx_env_open: MDBX_PANIC: Environment had fatal error, path:
581     snapshot_builder_test.go:387: readdirnames /var/folders/rq/h3d64x4x5kj9pcmhzt97q8yw0000gn/T/
582     testing.go:948: TempDir RemoveAll cleanup: openfdat /var/folders/rq/h3d64x4x5kj9pcmhzt97q8yw
583 FAIL
584 FAIL    github.com/ledgerwatch/erigon/turbo/snapshotsync    13.707s

**Comment by AskAlexSharov** [comment]
@BitBaseBit it's unstable test. I restarted CI.

**Comment by AskAlexSharov** [comment]
@BitBaseBit what do you think about switching from 
```
	defer func() { debug.LogPanic(nil, true, recover()) }()
```
to 
```
	defer debug.LogPanic(nil, true)
```
?

**Comment by AskAlexSharov** [comment]
@BitBaseBit , why do we pass error to LogPanic as a first param?

**Comment by AskAlexSharov** [comment]
How sending SIGINT behave inside tests? I mean if test call some high-level func which creating goroutine - and then this goroutine panics and self-SIGINT - will this test count as Failed?

**Pull request merged**